tests/std/tests/P0980R1_constexpr_strings/test.cpp

```diff
@@ -1754,60 +1825,339 @@ constexpr bool test_growth() {
+
+        assert(equalRanges(lhs, get_view_input<CharType>()));
+        assert(equalRanges(rhs, get_dog_view<CharType>()));
+
+        assert(lhs.get_allocator().id() == id2);
+        assert(rhs.get_allocator().id() == id1);
+    }
+
+    { // Allocated to Allocated
+        Str lhs(get_view_input<CharType>(), Alloc{id1});
+        Str rhs(get_view_input<CharType>(), Alloc{id2});
+        rhs.resize(30, 'a');
+        const auto lhs_begin = lhs.begin();
+        const auto rhs_begin = rhs.begin();
+
+        Str expected_lhs = rhs;
+
+        lhs.swap(rhs);
+        assert(lhs_begin == rhs.begin());
+        assert(rhs_begin == lhs.begin());
+
+        assert(equalRanges(lhs, expected_lhs));
+        assert(equalRanges(rhs, get_view_input<CharType>()));
+
+        assert(lhs.get_allocator().id() == id2);
+        assert(rhs.get_allocator().id() == id1);
+    }
+}
+
+template <class CharType>
+constexpr bool test_allocator_awareness() {
+    test_copy_ctor<CharType>();
+    test_copy_alloc_ctor<CharType>(11, 11); // equal allocators
+    test_copy_alloc_ctor<CharType>(11, 22); // non-equal allocators
+    test_copy_assign<CharType, StationaryAlloc<CharType>>(11, 11, 11); // non-POCCA, equal allocators
+    test_copy_assign<CharType, StationaryAlloc<CharType>>(11, 22, 22); // non-POCCA, non-equal allocators
+    test_copy_assign<CharType, CopyAlloc<CharType>>(11, 11, 11); // POCCA, equal allocators
+    test_copy_assign<CharType, CopyAlloc<CharType>>(11, 22, 11); // POCCA, non-equal allocators
+    test_copy_assign<CharType, CopyEqualAlloc<CharType>>(11, 22, 11); // POCCA, always-equal allocators
+
+    test_move_ctor<CharType>();
+    test_move_alloc_ctor<CharType>(11, 11); // equal allocators
+    test_move_alloc_ctor<CharType>(11, 22); // non-equal allocators
+
+    test_move_assign<CharType, StationaryAlloc<CharType>>(11, 11, 11); // non-POCMA, equal allocators
+    test_move_assign<CharType, StationaryAlloc<CharType>>(11, 22, 22); // non-POCMA, non-equal allocators
+    test_move_assign<CharType, MoveAlloc<CharType>>(11, 11, 11); // POCMA, equal allocators
+    test_move_assign<CharType, MoveAlloc<CharType>>(11, 22, 11); // POCMA, non-equal allocators
+    test_move_assign<CharType, MoveEqualAlloc<CharType>>(11, 22, 11); // POCMA, always-equal allocators
+
+    test_swap<CharType, StationaryAlloc<CharType>>(11, 11); // non-POCS, equal allocators
+    // UNDEFINED BEHAVIOR, NOT TESTED - non-POCS, non-equal allocators
+    test_swap<CharType, SwapAlloc<CharType>>(11, 11); // POCS, equal allocators
+    test_swap<CharType, SwapAlloc<CharType>>(11, 22); // POCS, non-equal allocators
+    test_swap<CharType, SwapEqualAlloc<CharType>>(11, 22); // POCS, always-equal allocators
+
+    return true;
+}
+
+template <class CharType>
+constexpr void test_all() {
+    test_interface<CharType>();
+    test_iterators<CharType>();
+    test_growth<CharType>();
+    test_allocator_awareness<CharType>();
+
+#ifndef __EDG__ // TRANSITION, VSO-1273296

```

Comment:
Is there a reason that the `#ifndef __EDG__ // TRANSITION, VSO-1273296` is being moved out of individual functions?  Is it just for ease/readability?  
I originally applied it with finer granularity around specific lines of code so we could have as much EDG coverage as possible for `constexpr string`.

Reply:
If I remember correctly, then @CaseyCarter was asking about some EDG special case I applied and it turned out to be fully disabled for EDG

Given that currently it is only citing a single bug I found it clearer to move the guard to the top

Reply:
Do you happen to recall where that conversation took place?  Looking through the activity in #1735 and here, I don't see anything to that effect.  I searched around in Discord a bit as well, but wasn't able to find this conversation.  

If possible, I would prefer to maintain a finer granularity for this switch (only disabling specific offending lines of code) so we have as much EDG test coverage as possible.  Given how many EDG bugs we encountered in `constexpr` containers, I'd like to continue to validate what we can so the fixes we _do_ have don't regress if the EDG team is working on fixes for the remaining issues.  

I'll approve the changes as this was my only concern, and perhaps we can leave this conversation "unresolved" and have the second reviewer chime in as well.

Reply:
I'll push changes to make the EDG workarounds fine-grained.